hathor/websocket/protocol.py

```diff
@@ -30,23 +35,144 @@ class HathorAdminWebsocketProtocol(WebSocketServerProtocol):
         can send the data update to the clients
     """
 
+    MAX_GAP_LIMIT: int = 30
+
     def __init__(self, factory: 'HathorAdminWebsocketFactory') -> None:
         self.log = logger.new()
         self.factory = factory
         self.subscribed_to: set[str] = set()
+        self._history_streamer: HistoryStreamer | None = None
+        self._manual_address_iter: ManualAddressSequencer | None = None
         super().__init__()
 
     def onConnect(self, request):
+        """Called by the websocket protocol when the connection is opened but it is still pending handshaking."""
         self.log.info('connection opened, starting handshake...', request=request)
 
     def onOpen(self) -> None:
+        """Called by the websocket protocol when the connection is established."""
         self.factory.on_client_open(self)
         self.log.info('connection established')
 
     def onClose(self, wasClean, code, reason):
+        """Called by the websocket protocol when the connection is closed."""
         self.factory.on_client_close(self)
         self.log.info('connection closed', reason=reason)
 
     def onMessage(self, payload: Union[bytes, str], isBinary: bool) -> None:
+        """Called by the websocket protocol when a new message is received."""
         self.log.debug('new message', payload=payload.hex() if isinstance(payload, bytes) else payload)
-        self.factory.handle_message(self, payload)
+        if isinstance(payload, bytes):
+            message = json_loadb(payload)
+        else:
+            message = json_loads(payload)
+
+        _type = message.get('type')
+
+        if _type == 'ping':
+            self._handle_ping(message)
+        elif _type == 'subscribe_address':
+            self.factory._handle_subscribe_address(self, message)
+        elif _type == 'unsubscribe_address':
+            self.factory._handle_unsubscribe_address(self, message)
+        elif _type == 'request:history:xpub':
+            self._open_history_xpub_streamer(message)
+        elif _type == 'request:history:manual':
```

Comment:
Maybe we should add a `request:history:abort` or `request:history:stop` so the client can warn the fullnode to stop streamming and not leave an open stream on the fullnode.

Reply:
I added a `request:history:stop`.